### DISCUSSION TITLE
Multi-Artist Feature

### DISCUSSION THREAD

**Discussion by SePHTaN** [ampache/ampache#2894] [Ideas]
@lachlan-00, @usox  
I've started to actually implement this in the last few days, and I'm not really happy with the approach I wanted to take first. Because I only thought of the possibility of multiple artists when it came to songs. But the same is possible with albums.

Therefore I have created a table showing which situations exist : 
Artists:
Artist		multiple not possible
Joinphrase	not needed as always single
Alias		        possible 			            ex.: Die Toten Hosen (primary)<-> Die Roten Rosen(alias)
Relationships?	possible 			            ex.: Cat Stevens     <-> Yusuf Islam

Albums:
Artist		multiple possible		    ex.: Cat Stevens & Elton John
Joinphrase 	needed if multiple artists	    ex.: &, feat.
Alias		        not needed / impossible?
Relationship?	not needed / impossible?

Songs:
Artist		multiple possible
Joinphrase	needed if multiple artists
Alias		        not needed / impossible?
Relationship	not needed / impossible?

My first idea for implementation was as following:
Artists:
Store artist as individual artist (all song artists)
Store artist aliases as indiviual artist with same mb_artistid (for aliases/pseudonym's)
for aliases artist table needs a field aliasof storing the artist_id of primary artist
remarks:
2 almost identical artists in database
still no relationships for artist that change like Cat Stevens -> Yusuf Islam are 2 Artists, 2 mbids but same Person

Albums:
haven't thought of before and wanted to leave as is (no good solution)

Songs:
add song for each artist (up to 6 song entry's according to my catalog for same song)
add joinphrase (for joining artistname as mentioned) and joinorder (for retaining joinorder) to song_data

So this feels not right / optimal for me because:
i have 1 artist but 2 database entries for him
i have 1 song but up to 6 (or even more) database entries

Suggestion:
Artists:
Only ONE database entry per Artist
Add Table Artist_data
Possible fields : Artist_aliases, and all other data where multiple values may exist

Albums:
Only ONE database entry per Album
Add Summary to table Album for album data like recension or similar
Add Table Album_data
Possible fields: mb_albumartistid, and all other data where multiple values may exist

Songs:
Only ONE database entry per Song
As with Artists and Albums all data that might have multiple values goes to song_data

Doing it this way the changes to the catalog doesn't interfere with the old implementation,
and migrating an old catalog to multi artist should be easier.

What are your opinions on this ?

**Comment by mitchray** [comment]
Multi-artist is in now so closing :)
